drivers/sensor/vl53l0x/vl53l0x.c

```diff
@@ -0,0 +1,254 @@
+/* vl53l0x.c - Driver for ST VL53L0X time of flight sensor */
+
+/*
+ * Copyright (c) 2017 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+
+#include <kernel.h>
+#include <i2c.h>
+#include <sensor.h>
+#include <init.h>
+#include <gpio.h>
+#include <misc/__assert.h>
+
+#include "vl53l0x.h"
+
+static int vl53l0x_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct vl53l0x_data *drv_data = (struct vl53l0x_data *)dev->driver_data;
```

Comment:
you don't need to cast driver_data explicitly as it is a void*, that's a detail though.